autogen/function_utils.py

```diff
@@ -266,7 +266,7 @@ def f(a: Annotated[str, "Parameter a"], b: int = 2, c: Annotated[float, "Paramet
         parameters=parameters,
     )
 
-    return model_dump(function)
+    return {"type": "function", "function": model_dump(function)}
```

Comment:
This is fine. I would prefer to create an additional Pydantic model and wrap the existing model in it. I can do it in a separate PR as it is a minor detail.

Reply:
@davorrunje how about let's do it here together?

Reply:
Ok, let me push it in a few minutes.

Reply:
New model in function_utils.py (after `class Function(BaseModel):`):
```python
class ToolFunction(BaseModel):
    """A function under tool as defined by the OpenAI API."""

    type: Literal["function"] = "function"
    function: Annotated[Function, Field(description="Function under tool")]
```
The new return value of `get_function_schema()`:
```python
    function = ToolFunction(
        function=Function(
            description=description,
            name=fname,
            parameters=parameters,
        )
    )

    return model_dump(function)
```

Reply:
@ekzhu @yenif I cannot push it, I don't have write access. But I tested it and it works.

Reply:
Awesome! I should be able to integrate, thanks!